### PR TITLE
Add Filter by DfTxType in AccountRPCs

### LINKED ISSUE #252: Filter `txtype` Parameter of `listAccountHistory()` doesn´t seem to filter AccountHistory
#### What happened:
I want to get a filtered list of transactions via `listAccountHistory`. I used the `txtype` options parameter which didn´t seem to have any effect. 

I tried different strings like `PoolSwap` or `Commission` with the same unfiltered result.

As this project is currently in heavy development the docs are limited at this point in time. I managed to find the following info:
`@param {string} [options.txtype] Filter by transaction type, supported letter from 'CRTMNnpuslrUbBG`

I have no clue what the letters suppose to mean. Can somebody explain?

An example from the tests shows a `txtype: 'M'` which in my case also didn´t filter the transactions.

#### What you expected to happen:

I expect to get only history objects of the `txtype` `Rewards` or `Commission` or `PoolSwap` or `...`.

#### How to reproduce it (as minimally and precisely as possible):

` "@defichain/jellyfish": "^0.3.0"`

```ts
import { Client } from "@defichain/jellyfish";
import { AccountHistoryOptions } from "@defichain/jellyfish-api-core/dist/category/account";

const client = new Client(
    "http://USER:PW@localhost:8555"
  );
  
  const options: AccountHistoryOptions = {
    txtype: "",
  };

  const accountHistories = await client.account.listAccountHistory(
    "mine",
    options
  );
```

### PR DESCRIPTION
**Pull request by siradji**
#### What kind of PR is this?:

/kind fix

#### What this PR does / why we need it:
Implemented `TxType` enum to fix #252
#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: packages/jellyfish-api-core/src/category/account.ts [hunk 1]
```diff
@@ -348,7 +348,7 @@ export interface AccountHistoryOptions {
   depth?: number
   no_rewards?: boolean
   token?: string
-  txtype?: string
+  txtype?: TxType | string
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/account.ts:351]: i think just `TxType` because it should be restricted, don't open for `string`

Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/account.ts:351]: i think just TxType because it should be restricted, we don't open for `string`

**Review by canonbrother — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/src/category/account.ts (hunk 1 above)
Comment (siradji) [reply packages/jellyfish-api-core/src/category/account.ts]: It's fixed.

File: website/docs/jellyfish/api/account.md [hunk 2]
```diff
@@ -125,6 +125,26 @@ enum OwnerType {
   ALL = "all"
 }
 
+enum TxType {
```
Comment (ivan-zynesis) [inline website/docs/jellyfish/api/account.md:128]:
```suggestion
enum DfTxType {
```

**Review by ivan-zynesis — CHANGES_REQUESTED** [review]
Would like to name it specifically apply for custom tx only

File: packages/jellyfish-api-core/src/category/account.ts [hunk 3]
```diff
@@ -285,7 +285,7 @@ export class Account {
    * @param {AccountHistoryCountOptions} [options]
    * @param {boolean} [options.no_rewards] Filter out rewards
    * @param {string} [options.token] Filter by token
-   * @param {TxType | string} [options.txtype] Filter by transaction type, supported letter from 'CRTMNnpuslrUbBG'
+   * @param {TxType} [options.txtype] Filter by transaction type, supported letter from 'CRTMNnpuslrUbBG'
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/src/category/account.ts:288]:
```suggestion
   * @param {TxType} [options.txtype] Filter by transaction type. See TxType.
```

For consistency

**Review by fuxingloh — DISMISSED** [review]

**Pull request closed without merging**